CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/time/LayerByTimePlugin.java

```diff
@@ -706,13 +710,13 @@ private void copyAttributes(final GraphWriteMethods graph, final int fromId, fin
      * Procedure for reordering the graph using nodes as layers on the z-axis.
      */
     private void nodesAsLayers(final GraphWriteMethods graph, final int txId, final int layer, final float normLayer) {
-        final int xAttr = graph.addAttribute(GraphElementType.VERTEX, "float", "x", "x", 0, null);
-        final int yAttr = graph.addAttribute(GraphElementType.VERTEX, "float", "y", "y", 0, null);
-        final int zAttr = graph.addAttribute(GraphElementType.VERTEX, "float", "z", "z", 0, null);
-        final int x2Attr = graph.addAttribute(GraphElementType.VERTEX, "float", "x2", "x2", 0, null);
-        final int y2Attr = graph.addAttribute(GraphElementType.VERTEX, "float", "y2", "y2", 0, null);
-        final int z2Attr = graph.addAttribute(GraphElementType.VERTEX, "float", "z2", "z2", 0, null);
-        final int vxVisibilityAttr = graph.addAttribute(GraphElementType.VERTEX, "float", "visibility", "visibility", 1, null);
+        final int xAttr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "x", "x", 0, null);
+        final int yAttr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "y", "y", 0, null);
+        final int zAttr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "z", "z", 0, null);
+        final int x2Attr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "x2", "x2", 0, null);
+        final int y2Attr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "y2", "y2", 0, null);
+        final int z2Attr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "z2", "z2", 0, null);
+        final int vxVisibilityAttr = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, VISIBILITY, VISIBILITY, 1, null);
```

Comment:
Why a constant for visibility here, but not x, y, z, ...?

Reply:
Because the rule didn't pick it up. It has less than 5 characters (tick) and/or has fewer than 3 occurences in the file. Given the sheer number of these I had to deal with, I opted not to give myself more work than needed (i.e. go out of my way to find ones not picked up)

Reply:
Up to you whether you want these as constants